build.zig

```diff
@@ -166,6 +178,226 @@ pub fn build(b: *Builder) !void {
     debug_step.dependOn(&debug_cmd.step);
 }
 
+/// The FAT32 step for creating a FAT32 image.
+const Fat32BuilderStep = struct {
+    /// The Step, that is all you need to know
+    step: Step,
+
+    /// The builder pointer, also all you need to know
+    builder: *Builder,
+
+    /// The file path to the bootstrap compiled code to be used in the FAT32 header.
+    fat32_bootstrap_bin: []const u8,
+
+    /// The path to where the ramdisk will be written to.
+    out_file_path: []const u8,
+
+    /// Options for creating the FAt32 image.
+    fat32_options: Options,
+
+    /// Options used for initialising a new FAT image.
+    const Options = struct {
+        /// A union for defining if a value has a user specified value or use the default value for
+        /// FAT32.
+        const DefaultOrValue = union(enum) {
+            /// The option will use the default value.
+            default,
+
+            /// A user provided value. This can be a unsigned integer or a string.
+            value: union {
+                int: usize,
+                str: []const u8,
+            },
+        };
+
+        /// The FAT32 image size in bytes. This is not the formatted size. Default see
+        /// `Fat32.getDefaultImageSize`
+        image_size: DefaultOrValue = .default,
```

Comment:
Instead of setting these to `.default`, could they just be set to the default value? Feel free to explain the reason why there's an extra level of indirection.

Reply:
That is a good point lol. I just went with it because I kinda like it ;)